cloudwatch_monitoring/api_calls.py

```diff
@@ -0,0 +1,198 @@
+                    MaxResults=10
+                )
+                response.update(response_iterator)
+            try:
+                next_token = response_iterator['NextToken']
+            except KeyError:
+                # no more pages, move on
+                break
+
+        return response
+
+    def is_iow_queue_filter(self, queue_url):
+        """
+        Apply filters to determine if the queue is a tagged IOW asset in the correct tier.
+
+        :param queue_url: A single queue's url
+        :return: is_iow_queue: is this an IOW queue or not
+        :rtype: bool
+        """
+        is_iow_queue = False
+
+        # filtering on deploy tier, which we capitalize
+        if self.deploy_stage.upper() in queue_url:
+            # launch API call to grab the tags for the queue
+            queue_tags = self.boto3_client.list_queue_tags(QueueUrl=queue_url)
+
+            # we only want queues that are tagged as 'IOW'
+            if 'Tags' in queue_tags:
+                if 'wma:organization' in queue_tags['Tags']:
+                    if 'IOW' == queue_tags['Tags']['wma:organization']:
+                        is_iow_queue = True
+
+        return is_iow_queue
+
+    def get_all_lambda_metadata(self):
+        """
+        Using the AWS python sdk (boto3), grab all the lambda functions for the specified account for a given region.
+
+        :return: response: metadata about each lambda in the account.
+        :rtype: dict
+        """
+
+        # Currently you cannot get more than 50 functions from the list_functions call in a single request.  Thus, we need
+        # to iterate over the entire list of available functions in the account using the provided NextMarker string, which
+        # allows us to paginate. boto3 does have pagination tools, but I have found the documentation generally unhelpful,
+        # so below is a more manual approach.
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda.html#Lambda.Client.list_functions
+        # TODO this pagination logic exists in the sqs module as well, consider moving it into its own utility
+        # TODO module or trying to get a proper boto3 paginator to work...
+        response = {}
+        marker = None
+        while True:
+            if marker:
+                response_iterator = self.boto3_client.list_functions(
+                    MaxItems=10,
+                    Marker=marker)
+                response['Functions'].extend(response_iterator['Functions'])
+            else:
+                response_iterator = self.boto3_client.list_functions(
+                    MaxItems=10
+                )
+                response.update(response_iterator)
+            try:
+                marker = response_iterator['NextMarker']
+            except KeyError:
+                # no more pages, move on
+                break
+
+        return response
+
+    def is_iow_lambda_filter(self, function):
```

Comment:
I'm not sure I would mix different types of the AWS services within the same class. This works, but when an object is instantiated, the boto3 client that's created will only work one for a subset of the methods.